default-plugin/src/main/java/me/shedaniel/rei/plugin/client/categories/DefaultCompostingCategory.java

```diff
@@ -95,15 +94,15 @@ public List<Widget> setupDisplay(DefaultCompostingDisplay display, Rectangle bou
         int i = 0;
         for (int y = 0; y < 5; y++)
             for (int x = 0; x < 7; x++) {
-                EntryIngredient entryStack = stacks.size() > i ? stacks.get(i) : EntryIngredient.empty();
-                if (!entryStack.isEmpty()) {
-                    ComposterBlock.COMPOSTABLES.object2FloatEntrySet().stream().filter(entry -> entry.getKey() != null && Objects.equals(entry.getKey().asItem(), entryStack.get(0).getValue())).findAny().map(Map.Entry::getValue).ifPresent(chance -> {
-                        for (EntryStack<?> stack : entryStack) {
-                            stack.tooltip(Component.translatable("text.rei.composting.chance", Mth.fastFloor(chance * 100)).withStyle(ChatFormatting.YELLOW));
-                        }
-                    });
+                EntryIngredient entryIngredient = stacks.size() > i ? stacks.get(i) : EntryIngredient.empty();
+                if (!entryIngredient.isEmpty()) {
+                    ItemStack firstStack = (ItemStack) entryIngredient.get(0).getValue();
+                    float chance = ComposterBlock.COMPOSTABLES.object2FloatEntrySet().stream().filter(entry -> entry.getKey() != null && firstStack.is(entry.getKey().asItem())).findAny().map(Map.Entry::getValue).orElse(0f);
```

Comment:
Is it possible to turn this into O(1)?

Reply:
i literally just realized how dumb this line is when i can literally just query `COMPOSTABLES.getFloat(firstStack.getItem())`
updated